File: src/main/kotlin/com/example/swip/service/SecondChapter/GroovyCompiler.java
package com.example.swip.service.SecondChapter;

import groovy.lang.Binding;
import groovy.lang.GroovyShell;

public class GroovyCompiler {
    public GroovyCompiler() {
    }

    public String getAnswer2_1(String userData){
        Binding binding = new Binding();
        GroovyShell shell = new GroovyShell(binding);

        return String.valueOf(shell.evaluate("System.out.println(\"Привет вот данные\" + userData);"));
    }

    public static void main(String[] args) {
        /*Binding binding = new Binding();
        GroovyShell shell = new GroovyShell(binding);

        shell.evaluate("public class FirstSample\n" +
                "{\n" +
                "   public static void main(String[] args)\n" +
                "   {\n" +
                "      System.out.println(\"We wid!'\");\n" +
                "   \n" +
                "}");*/
    }
}

File: src/main/kotlin/com/example/swip/service/groovyCompiler/GroovyCompiler.kt
package com.example.swip.service.groovyCompiler

import groovy.lang.GroovyShell
import org.springframework.stereotype.Service
import java.io.ByteArrayOutputStream
import java.io.PrintStream

@Service
class GroovyCompiler {
    fun execute(codeSTR: String): String {
        val binding = groovy.lang.Binding()
        val shell = GroovyShell(binding)

        //перенаправим потоки вывода
        val buffer = ByteArrayOutputStream()
        val saveSystemOut = System.out
        System.setOut(PrintStream(buffer))

        try {
            shell.evaluate(codeSTR)
        } catch (e: Error) {
            System.setOut(saveSystemOut)
            return "Incorrect answer"
        } catch (e: Exception) {
            System.setOut(saveSystemOut)
            return "Incorrect answer"
        }

        //вернем все на место
        System.setOut(saveSystemOut)

        val compiledValue = buffer.toString()
        return compiledValue.subSequence(0, compiledValue.length - 2).toString()
    }
}